lib/ressipy/web/controllers/auth_controller.ex

```diff
@@ -22,46 +21,38 @@ defmodule Ressipy.Web.AuthController do
     |> redirect(to: "/")
   end
 
-  defp create_user(%{uid: facebook_id, info: %{name: name}}) do
-    role_name = Application.get_env(:ressipy, :default_role)
-    role_id = case Repo.all from role in Role, where: role.name == ^role_name do
-      [%Role{id: id} | _] -> id
-      _ -> raise "Could not find a role named #{role_name}"
-    end
-    %{name: name, facebook_id: facebook_id, role_id: role_id}
+  defp create_user_if_needed(nil, %{uid: facebook_id, info: %{name: name}}) do
+    %{name: name, facebook_id: facebook_id}
     |> Accounts.create_user
   end
 
+  defp create_user_if_needed(%User{} = user, _) do
+    {:ok, user}
+  end
+
   def delete(conn, _params) do
     conn
     |> put_flash(:info, "You have been logged out!")
     |> configure_session(drop: true)
     |> redirect(to: "/")
   end
 
-  defp find_user(auth) do
-    case Repo.all from user in User,
-      where: user.facebook_id == ^auth.uid,
-      preload: [:role] do
-      [%User{} = user | _] -> user
-      _ -> nil
-    end
+  defp find_user(%Ueberauth.Auth{uid: facebook_id}) do
+    Repo.one from u in User,
+      where: u.facebook_id == ^facebook_id,
+      preload: :role,
+      limit: 1
```

Comment:
Instead of running this as two separate queries, let's do:
```
Repo.one from u in User,
  join: r in assoc(u, :role),
  preload: [role: r],
  limit: 1
```